src/luarocks/cmd/unpack.lua

```diff
@@ -21,126 +16,6 @@ In the latter case, the app version may be given as a second argument.
-local function unpack_rock(rock_file, dir_name, kind)
-   assert(type(rock_file) == "string")
-   assert(type(dir_name) == "string")
-
-   local ok, err, errcode = fetch.fetch_and_unpack_rock(rock_file, dir_name)
-   if not ok then
-      return nil, "Failed unzipping rock "..rock_file, errcode
-   end
-   ok, err = fs.change_dir(dir_name)
-   if not ok then return nil, err end
-   local rockspec_file = dir_name..".rockspec"
-   local rockspec, err = fetch.load_rockspec(rockspec_file)
-   if not rockspec then
-      return nil, "Failed loading rockspec "..rockspec_file..": "..err
-   end
-   if kind == "src" then
-      if rockspec.source.file then
-         local ok, err = fs.unpack_archive(rockspec.source.file)
-         if not ok then
-            return nil, err
-         end
-         ok, err = fs.change_dir(rockspec.source.dir)
-         if not ok then return nil, err end
-         ok, err = build.apply_patches(rockspec)
-         fs.pop_dir()
-         if not ok then return nil, err end
-      end
-   end
-   return rockspec
-end
-
---- Create a directory and perform the necessary actions so that
--- the sources for the rock and its rockspec are unpacked inside it,
--- laid out properly so that the 'make' command is able to build the module.
--- @param file string: A rockspec or .rock URL.
--- @return boolean or (nil, string): true if successful or nil followed
--- by an error message.
-local function run_unpacker(file, force)
-   assert(type(file) == "string")
-   
-   local base_name = dir.base_name(file)
-   local dir_name, kind, extension = base_name:match("(.*)%.([^.]+)%.(rock)$")
-   if not extension then
-      dir_name, extension = base_name:match("(.*)%.(rockspec)$")
-      kind = "rockspec"
-   end
-   if not extension then
-      return nil, file.." does not seem to be a valid filename."
-   end
-
-   local exists = fs.exists(dir_name)
-   if exists and not force then
-      return nil, "Directory "..dir_name.." already exists."
-   end
-   if not exists then
-      local ok, err = fs.make_dir(dir_name)
-      if not ok then return nil, err end
-   end
-   local rollback = util.schedule_function(fs.delete, fs.absolute_name(dir_name))
-
-   local rockspec, err
-   if extension == "rock" then
-      rockspec, err = unpack_rock(file, dir_name, kind)
-   elseif extension == "rockspec" then
-      rockspec, err = unpack_rockspec(file, dir_name)
-   end
-   if not rockspec then
-      return nil, err
-   end
-   if kind == "src" or kind == "rockspec" then
-      if rockspec.source.dir ~= "." then
-         local ok = fs.copy(rockspec.local_abs_filename, rockspec.source.dir, "read")
-         if not ok then
-            return nil, "Failed copying unpacked rockspec into unpacked source directory."
-         end
-      end
-      cmd.printout()   
-      cmd.printout("Done. You may now enter directory ")
-      cmd.printout(dir.path(dir_name, rockspec.source.dir))
-      cmd.printout("and type 'luarocks make' to build.")
```

Comment:
Looks like these messages were lost in the conversion — these could remain in the `luarocks.cmd.unpack` module. One way to do this would be to return `kind, nil, dir.path(dir_name, rockspec.source.dir)` on success, and then in `unpack.command` to something like

```lua
local kind, err, srcdir = luarocks.unpack(ns_name, flags["force"])
if not kind then
   return nil, err
end
if kind == "src" or kind == "rockspec" then
   cfg.log()
   cfg.log("Done. You may now enter directory ")
   cfg.log(srcdir)
   cfg.log("and type 'luarocks make' to build.")
end
return true
```

Reply:
Since these messages are only relevant when using `luarocks` from the terminal, I kept using `cmd.printout` instead of `cfg.log`.

Reply:
@georgeroman Makes total sense!